megamek/src/megamek/client/ui/swing/tileset/MMStaticDirectoryManager.java

```diff
@@ -0,0 +1,537 @@
+/*
+ * Copyright (c) 2020 - The MegaMek Team. All Rights Reserved.
+ *
+ * This file is part of MegaMek.
+ *
+ * MegaMek is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * MegaMek is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MegaMek. If not, see <http://www.gnu.org/licenses/>.
+ */
+package megamek.client.ui.swing.tileset;
+
+import megamek.MegaMek;
+import megamek.client.ui.swing.util.ImageFileFactory;
+import megamek.client.ui.swing.util.PlayerColors;
+import megamek.client.ui.swing.util.ScaledImageFileFactory;
+import megamek.common.Configuration;
+import megamek.common.Crew;
+import megamek.common.Entity;
+import megamek.common.IPlayer;
+import megamek.common.annotations.Nullable;
+import megamek.common.util.ImageUtil;
+import megamek.common.util.fileUtils.DirectoryItem;
+import megamek.common.util.fileUtils.DirectoryItems;
+
+import javax.swing.*;
+import java.awt.*;
+import java.awt.image.BufferedImage;
+
+public class MMStaticDirectoryManager {
+    //region Variable Declarations
+    // Directories
+    private static DirectoryItems portraitDirectory;
+    private static DirectoryItems camouflageDirectory;
+    private static MechTileset mechTileset;
+
+    // Re-parsing Prevention Variables: The are True at startup and when the specified directory
+    // should be re-parsed, and are used to avoid re-parsing the directory repeatedly when there's an error.
+    private static boolean parsePortraitDirectory = true;
+    private static boolean parseCamouflageDirectory = true;
+    private static boolean parseMechTileset = true;
+    //endregion Variable Declarations
+
+    //region Constructors
+    protected MMStaticDirectoryManager() {
+        // This class is not to be instantiated
+    }
+    //endregion Constructors
+
+    //region Initialization
+    /**
+     * This initialized all of the directories under this manager
+     */
+    public static void initialize() {
+        initializePortraits();
+        initializeCamouflage();
+        initializeMechTileset();
+    }
+
+    /**
+     * Parses MM's portraits folder when first called or when it was refreshed.
+     *
+     * @see #refreshPortraitDirectory()
+     */
+    private static void initializePortraits() {
```

Comment:
Should these initializers should be synchronized? (If so, not `initialize` itself, to avoid deadlocks).

Reply:
There haven't been any issues with it as of yet, but that may become an issue in future. Maybe @SJuliez can comment on why it wasn't done that way originally?

Reply:
I didn't think about it. I think MM uses only few extra threads, for the RATs, boardview refresh and pathfinding. None of these will request an image. So perhaps we are safe now...